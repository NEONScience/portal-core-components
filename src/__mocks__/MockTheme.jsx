import React from 'react';

import CssBaseline from '@mui/material/CssBaseline';
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles';

import Theme from '../lib_components/components/Theme/Theme';

const MockTheme = ({ children }) => {
  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={Theme}>
        <CssBaseline />
        {children}
      </ThemeProvider>
    </StyledEngineProvider>
  );
}

export default MockTheme;
