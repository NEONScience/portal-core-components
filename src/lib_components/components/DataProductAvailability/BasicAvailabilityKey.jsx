import React, {
  useState,
  useRef,
  useLayoutEffect,
  useCallback,
} from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from '@material-ui/core/styles';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';

import CloseIcon from '@material-ui/icons/Close';
import HelpIcon from '@material-ui/icons/HelpOutline';

import { SVG, VALID_ENHANCED_STATUSES } from './AvailabilityUtils';
import { JsxCell } from './AvailabilitySvgComponents';

import Theme, { COLORS } from '../Theme/Theme';
import { exists } from '../../util/typeUtil';

const useStyles = makeStyles((theme) => ({
  legendContainer: {
    display: 'flex',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    flexWrap: 'wrap',
    marginTop: theme.spacing(1),
    width: '100%',
  },
  legendTitle: {
    fontSize: '0.95rem',
    display: 'inline-flex',
  },
  legendTitleIcon: {
    marginTop: theme.spacing(-0.25),
  },
  legendTitleContainer: {
    margin: theme.spacing(-1, 1, 0.5, 0),
  },
  legendElement: {
    margin: theme.spacing(0, 0, 0, 0),
  },
  legendElementText: {
    textAnchor: 'start',
    whiteSpace: 'pre',
    fontFamily: '"Cutive Mono","Lucida Console",Monaco,monospace',
    fontWeight: 400,
    fontSize: `${SVG.LABEL_FONT_SIZE}px`,
    fill: Theme.palette.grey[700],
  },
}));

const CONTAINER_WIDTH_BREAKPOINT_XS_FLEX_COL = 250;
const CONTAINER_WIDTH_BREAKPOINT_XS = 360;
const CONTAINER_WIDTH_BREAKPOINT_XS_SELECTION = 200;
const CONTAINER_WIDTH_BREAKPOINT_SM_SELECTION = 425;

const ALL_SELECTED_TITLE = 'If the chart is presenting a roll-up (e.g. view by state) then all sites rolled into a given row are selected';
const SOME_SELECTED_TITLE = 'If the chart is presenting a roll-up (e.g. view by state) then one or more but not all of the sites rolled into a given row are selected';

const StatusLegendElement = (props) => {
  const classes = useStyles(Theme);
  const { status, dialog } = props;
  if (!exists(status) || !VALID_ENHANCED_STATUSES[status]) {
    return null;
  }
  const statusSvgHeight = SVG.CELL_HEIGHT + 2;
  const labelLetterWidth = 8;
  const labelY = SVG.LABEL_FONT_SIZE - SVG.CELL_PADDING + 2;
  const statusLabelX = SVG.CELL_WIDTH + (2 * SVG.CELL_PADDING);
  const { title, description } = VALID_ENHANCED_STATUSES[status];
  const statusSvgWidth = (title.length * labelLetterWidth) + statusLabelX;
  return dialog ? (
    <div style={{ marginBottom: Theme.spacing(2.5) }}>
      <div style={{ display: 'flex', alignItems: 'center' }}>
        <svg
          width={Math.ceil(SVG.CELL_WIDTH * 1.25)}
          height={Math.ceil(SVG.CELL_HEIGHT * 1.25)}
          viewBox={`0 0 ${SVG.CELL_WIDTH} ${SVG.CELL_HEIGHT}`}
          style={{ marginRight: Theme.spacing(1) }}
        >
          <JsxCell status={status} />
        </svg>
        <Typography variant="subtitle2" style={{ fontSize: '0.95rem', marginTop: '2px' }}>
          {title}
        </Typography>
      </div>
      <Typography variant="body2">{description}</Typography>
    </div>
  ) : (
    <div className={classes.legendElement} title={description}>
      <svg width={statusSvgWidth} height={statusSvgHeight}>
        <JsxCell status={status} />
        <text className={classes.legendElementText} x={statusLabelX} y={labelY}>
          {title}
        </text>
      </svg>
    </div>
  );
};

StatusLegendElement.propTypes = {
  status: PropTypes.oneOf(Object.keys(VALID_ENHANCED_STATUSES)),
  dialog: PropTypes.bool,
};

StatusLegendElement.defaultProps = {
  status: null,
  dialog: false,
};

const SelectionLegendElement = (props) => {
  const classes = useStyles(Theme);
  const { variant, dialog } = props;
  if (!['all', 'some'].includes(variant)) {
    return null;
  }
  const statusSvgHeight = SVG.CELL_HEIGHT + 2;
  const labelLetterWidth = 8;
  const labelY = SVG.LABEL_FONT_SIZE - SVG.CELL_PADDING + 2;
  const selectionSvgHeight = SVG.CELL_HEIGHT + 2;
  const label = variant === 'all' ? 'All sites selected' : 'Some sites selected';
  const fill = variant === 'all' ? Theme.palette.primary.main : COLORS.LIGHT_BLUE[200];
  const description = variant === 'all' ? ALL_SELECTED_TITLE : SOME_SELECTED_TITLE;
  const selectionWidth = 30;
  const selectionLabelX = selectionWidth + (3 * SVG.CELL_PADDING);
  const selectionSvgWidth = (label.length * labelLetterWidth) + selectionLabelX;
  const handleAttribs = {
    width: SVG.DATE_RANGE_HANDLE_WIDTH,
    height: SVG.CELL_HEIGHT,
    fill: COLORS.LIGHT_BLUE[300],
    stroke: Theme.palette.primary.main,
    strokeWidth: 1.5,
  };
  const graphic = (
    <>
      <rect x={0.5} y={1.5} width={selectionWidth} height={SVG.CELL_HEIGHT - 2} fill={fill} />
      <rect x={0.5} y={0.5} {...handleAttribs} />
      <rect x={selectionWidth - SVG.DATE_RANGE_HANDLE_WIDTH} y={0.5} {...handleAttribs} />
    </>
  );
  return dialog ? (
    <div style={{ marginBottom: Theme.spacing(2) }}>
      <div style={{ display: 'flex', alignItems: 'center' }}>
        <svg
          width={Math.ceil(selectionWidth * 1.25)}
          height={Math.ceil(selectionSvgHeight * 1.25)}
          viewBox={`0 0 ${selectionWidth} ${selectionSvgHeight}`}
          style={{ marginRight: Theme.spacing(1) }}
        >
          {graphic}
        </svg>
        <Typography variant="subtitle2" style={{ fontSize: '1.05rem' }}>{label}</Typography>
      </div>
      <Typography variant="body2">{description}</Typography>
    </div>
  ) : (
    <div className={classes.legendElement} title={description}>
      <svg width={selectionSvgWidth} height={statusSvgHeight}>
        {graphic}
        <text className={classes.legendElementText} x={selectionLabelX} y={labelY}>
          {label}
        </text>
      </svg>
    </div>
  );
};

SelectionLegendElement.propTypes = {
  variant: PropTypes.oneOf(['all', 'some']).isRequired,
  dialog: PropTypes.bool,
};

SelectionLegendElement.defaultProps = {
  dialog: false,
};

const LegendDialog = (props) => {
  const {
    dialogOpen,
    setDialogOpen,
    selectionEnabled,
    delineateRelease,
    availabilityStatusType,
  } = props;
  return (
    <Dialog
      open={dialogOpen}
      maxWidth="md"
      onClose={() => setDialogOpen(false)}
      aria-labelledby="availability-key-dialog-title"
    >
      <div style={{ display: 'flex', alignItems: 'center', justifyContent: 'space-between' }}>
        <DialogTitle id="availability-key-dialog-title">
          <span style={{ fontSize: '1.4rem', fontWeight: '600' }}>
            Data Availability Chart Key
          </span>
        </DialogTitle>
        <IconButton
          title="Close"
          aria-label="Close"
          onClick={() => setDialogOpen(false)}
          style={{ marginRight: Theme.spacing(1) }}
        >
          <CloseIcon fontSize="inherit" />
        </IconButton>
      </div>
      <DialogContent style={{ marginBottom: Theme.spacing(2) }}>
        <Grid container spacing={2}>
          <Grid item xs={12} sm={6}>
            <StatusLegendElement status={availabilityStatusType} dialog />
          </Grid>
          {!delineateRelease ? null : (
            <Grid item xs={12} sm={6}>
              <StatusLegendElement status="available-provisional" dialog />
            </Grid>
          )}
          <Grid item xs={12} sm={6}>
            <StatusLegendElement status="not available" dialog />
          </Grid>
          {!delineateRelease ? null : (
            <Grid item xs={12} sm={6}>
              <StatusLegendElement status="mixed-available-provisional" dialog />
            </Grid>
          )}
          {!selectionEnabled ? null : (
            <>
              <Grid item xs={12} sm={6}>
                <SelectionLegendElement variant="all" dialog />
              </Grid>
              <Grid item xs={12} sm={6}>
                <SelectionLegendElement variant="some" dialog />
              </Grid>
            </>
          )}
        </Grid>
      </DialogContent>
    </Dialog>
  );
};

LegendDialog.propTypes = {
  dialogOpen: PropTypes.bool.isRequired,
  setDialogOpen: PropTypes.func.isRequired,
  selectionEnabled: PropTypes.bool,
  delineateRelease: PropTypes.bool,
  availabilityStatusType: PropTypes.oneOf(['available', 'tombstoned']),
};

LegendDialog.defaultProps = {
  selectionEnabled: false,
  delineateRelease: false,
  availabilityStatusType: 'available',
};

const BasicAvailabilityKey = (props) => {
  const classes = useStyles(Theme);
  const {
    selectionEnabled,
    delineateRelease,
    availabilityStatusType,
    dialogOnly,
  } = props;
  const appliedAvaStatusType = !exists(availabilityStatusType)
    ? 'available'
    : availabilityStatusType;
  const containerRef = useRef();
  const [dialogOpen, setDialogOpen] = useState(false);
  const [componentWidth, setComponentWidth] = useState(0);
  let atContainerWidthXsFlexCol = false;
  let atContainerWidthXs = false;
  let atContainerWidthSelectionXs = false;
  let atContainerWidthSelectionSm = false;
  if (componentWidth > 0) {
    atContainerWidthXsFlexCol = (componentWidth <= CONTAINER_WIDTH_BREAKPOINT_XS_FLEX_COL);
    atContainerWidthXs = (componentWidth <= CONTAINER_WIDTH_BREAKPOINT_XS);
    atContainerWidthSelectionXs = (componentWidth <= CONTAINER_WIDTH_BREAKPOINT_XS_SELECTION);
    atContainerWidthSelectionSm = (componentWidth <= CONTAINER_WIDTH_BREAKPOINT_SM_SELECTION);
  }
  const containerStyleProps = atContainerWidthXsFlexCol
    ? { flexDirection: 'column' }
    : {};
  const handleResizeCb = useCallback(() => {
    const container = containerRef.current;
    if (!container) { return; }
    if (container.clientWidth === componentWidth) { return; }
    setComponentWidth(container.clientWidth);
  }, [containerRef, componentWidth, setComponentWidth]);
  useLayoutEffect(() => {
    const element = containerRef.current;
    if (!element) { return () => {}; }
    handleResizeCb();
    if (typeof ResizeObserver !== 'function') {
      window.addEventListener('resize', handleResizeCb);
      return () => {
        window.removeEventListener('resize', handleResizeCb);
      };
    }
    let resizeObserver = new ResizeObserver(handleResizeCb);
    resizeObserver.observe(element);
    return () => {
      if (!resizeObserver) { return; }
      resizeObserver.disconnect();
      resizeObserver = null;
    };
  }, [containerRef, handleResizeCb]);

  const renderLegendItems = () => {
    if (dialogOnly) return null;
    if (selectionEnabled) {
      return (
        <>
          <div>
            <StatusLegendElement status={appliedAvaStatusType} dialog={false} />
            {!delineateRelease ? null : (
              <StatusLegendElement status="available-provisional" dialog={false} />
            )}
            <StatusLegendElement status="not available" dialog={false} />
            {!delineateRelease ? null : (
              <StatusLegendElement status="mixed-available-provisional" dialog={false} />
            )}
          </div>
          {!selectionEnabled ? null : (
            <div>
              <SelectionLegendElement variant="all" dialog={false} />
              <SelectionLegendElement variant="some" dialog={false} />
            </div>
          )}
        </>
      );
    }
    return (
      <>
        <div>
          <StatusLegendElement status={appliedAvaStatusType} dialog={false} />
          {!delineateRelease ? null : (
            <StatusLegendElement status="available-provisional" dialog={false} />
          )}
        </div>
        <div>
          <StatusLegendElement status="not available" dialog={false} />
          {!delineateRelease ? null : (
            <StatusLegendElement status="mixed-available-provisional" dialog={false} />
          )}
        </div>
      </>
    );
  };

  const renderLegendItemsContainer = () => {
    if (dialogOnly || atContainerWidthSelectionXs) return null;
    if (selectionEnabled) {
      if (atContainerWidthSelectionSm) {
        // If at sm selection container width, wrap in containing div
        // to display all statuses vertically in single row
        return (<div>{renderLegendItems()}</div>);
      }
      return renderLegendItems();
    }
    // If at xs container width, wrap in containing div
    // to display all statuses vertically in single row
    if (atContainerWidthXs) {
      return (<div>{renderLegendItems()}</div>);
    }
    return renderLegendItems();
  };

  return (
    <>
      <div
        ref={containerRef}
        className={classes.legendContainer}
        style={containerStyleProps}
      >
        <div className={classes.legendTitleContainer}>
          <Typography variant="h6" className={classes.legendTitle}>
            Key:
          </Typography>
          <IconButton
            size="small"
            color="primary"
            title="Help - Data Availability Chart Key"
            aria-label="Help - Data Availability Chart Key"
            onClick={() => setDialogOpen(true)}
            className={classes.legendTitleIcon}
          >
            <HelpIcon fontSize="small" />
          </IconButton>
        </div>
        {renderLegendItemsContainer()}
      </div>
      <LegendDialog
        dialogOpen={dialogOpen}
        setDialogOpen={setDialogOpen}
        selectionEnabled={selectionEnabled}
        delineateRelease={delineateRelease}
        availabilityStatusType={appliedAvaStatusType}
      />
    </>
  );
};

BasicAvailabilityKey.propTypes = {
  selectionEnabled: PropTypes.bool,
  delineateRelease: PropTypes.bool,
  availabilityStatusType: PropTypes.oneOf(['available', 'tombstoned']),
  dialogOnly: PropTypes.bool,
};

BasicAvailabilityKey.defaultProps = {
  selectionEnabled: false,
  delineateRelease: false,
  availabilityStatusType: 'available',
  dialogOnly: false,
};

export default BasicAvailabilityKey;
