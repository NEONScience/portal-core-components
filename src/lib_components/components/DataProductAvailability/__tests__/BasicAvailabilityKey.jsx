import React from 'react';
import renderer from 'react-test-renderer';

import BasicAvailabilityKey from '../BasicAvailabilityKey';

import MockTheme from '../../../../__mocks__/MockTheme';

describe('DataProductAvailability - BasicAvailabilityKey', () => {
  test('renders correctly when selection is not enabled (default)', () => {
    const tree = renderer
      .create(<MockTheme><BasicAvailabilityKey /></MockTheme>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders correctly when selection is enabled', () => {
    const tree = renderer
      .create(<MockTheme><BasicAvailabilityKey selectionEnabled /></MockTheme>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
