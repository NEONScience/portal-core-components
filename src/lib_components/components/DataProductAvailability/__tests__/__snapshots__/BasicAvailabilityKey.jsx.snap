// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataProductAvailability - BasicAvailabilityKey renders correctly when selection is enabled 1`] = `
<div
  className="makeStyles-legendContainer-1"
>
  <div
    className="makeStyles-legendTitleContainer-4"
  >
    <h6
      className="MuiTypography-root makeStyles-legendTitle-2 MuiTypography-h6"
    >
      Key:
    </h6>
    <button
      aria-label="Help - Data Availability Chart Key"
      className="MuiButtonBase-root MuiIconButton-root makeStyles-legendTitleIcon-3 MuiIconButton-colorPrimary MuiIconButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Help - Data Availability Chart Key"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div>
    <div
      className="makeStyles-legendElement-5"
      title="Data have been published and released. Data are available for download"
    >
      <svg
        height={14}
        width={151}
      >
        <rect
          fill="#002c77"
          height="12px"
          rx="2px"
          stroke={null}
          strokeWidth={null}
          width="7px"
          x={0}
          y={0}
        />
        <text
          className="makeStyles-legendElementText-6"
          x={15}
          y={10}
        >
          Release Available
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-legendElement-5"
      title="Data was neither expected nor collected"
    >
      <svg
        height={14}
        width={119}
      >
        <rect
          fill="#d7d9d9"
          height="12px"
          rx="2px"
          stroke={null}
          strokeWidth={null}
          width="7px"
          x={0}
          y={0}
        />
        <text
          className="makeStyles-legendElementText-6"
          x={15}
          y={10}
        >
          Not Available
        </text>
      </svg>
    </div>
  </div>
  <div>
    <div
      className="makeStyles-legendElement-5"
      title="If the chart is presenting a roll-up (e.g. view by state) then all sites rolled into a given row are selected"
    >
      <svg
        height={14}
        width={186}
      >
        <rect
          fill="#0073cf"
          height={10}
          width={30}
          x={0.5}
          y={1.5}
        />
        <rect
          fill="#5ca6e3"
          height={12}
          stroke="#0073cf"
          strokeWidth={1.5}
          width={4}
          x={0.5}
          y={0.5}
        />
        <rect
          fill="#5ca6e3"
          height={12}
          stroke="#0073cf"
          strokeWidth={1.5}
          width={4}
          x={26}
          y={0.5}
        />
        <text
          className="makeStyles-legendElementText-6"
          x={42}
          y={10}
        >
          All sites selected
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-legendElement-5"
      title="If the chart is presenting a roll-up (e.g. view by state) then one or more but not all of the sites rolled into a given row are selected"
    >
      <svg
        height={14}
        width={194}
      >
        <rect
          fill="#8abfec"
          height={10}
          width={30}
          x={0.5}
          y={1.5}
        />
        <rect
          fill="#5ca6e3"
          height={12}
          stroke="#0073cf"
          strokeWidth={1.5}
          width={4}
          x={0.5}
          y={0.5}
        />
        <rect
          fill="#5ca6e3"
          height={12}
          stroke="#0073cf"
          strokeWidth={1.5}
          width={4}
          x={26}
          y={0.5}
        />
        <text
          className="makeStyles-legendElementText-6"
          x={42}
          y={10}
        >
          Some sites selected
        </text>
      </svg>
    </div>
  </div>
</div>
`;

exports[`DataProductAvailability - BasicAvailabilityKey renders correctly when selection is not enabled (default) 1`] = `
<div
  className="makeStyles-legendContainer-1"
>
  <div
    className="makeStyles-legendTitleContainer-4"
  >
    <h6
      className="MuiTypography-root makeStyles-legendTitle-2 MuiTypography-h6"
    >
      Key:
    </h6>
    <button
      aria-label="Help - Data Availability Chart Key"
      className="MuiButtonBase-root MuiIconButton-root makeStyles-legendTitleIcon-3 MuiIconButton-colorPrimary MuiIconButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Help - Data Availability Chart Key"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
          />
        </svg>
      </span>
    </button>
  </div>
  <div>
    <div
      className="makeStyles-legendElement-5"
      title="Data have been published and released. Data are available for download"
    >
      <svg
        height={14}
        width={151}
      >
        <rect
          fill="#002c77"
          height="12px"
          rx="2px"
          stroke={null}
          strokeWidth={null}
          width="7px"
          x={0}
          y={0}
        />
        <text
          className="makeStyles-legendElementText-6"
          x={15}
          y={10}
        >
          Release Available
        </text>
      </svg>
    </div>
  </div>
  <div>
    <div
      className="makeStyles-legendElement-5"
      title="Data was neither expected nor collected"
    >
      <svg
        height={14}
        width={119}
      >
        <rect
          fill="#d7d9d9"
          height="12px"
          rx="2px"
          stroke={null}
          strokeWidth={null}
          width="7px"
          x={0}
          y={0}
        />
        <text
          className="makeStyles-legendElementText-6"
          x={15}
          y={10}
        >
          Not Available
        </text>
      </svg>
    </div>
  </div>
</div>
`;
